samples/bpf/xdplua_user.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) 2019-2020 Victor Nogueira <victor.nogueira@ring-0.io>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation version 2.
+ *
+ * This program is distributed "as is" WITHOUT ANY WARRANTY of any
+ * kind, whether express or implied; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+#include <linux/if_link.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <libgen.h>
+#include <sys/resource.h>
+#include <net/if.h>
+
+#include <bpf/bpf.h>
+#include "bpf/libbpf.h"
+
+#include "bpf_util.h"
+
+#define MAXSCRIPTLEN 8192
+#define MAXFILENAMELEN 256
+
+static int ifindex = 0;
+
+static void usage(const char *prog) {
+	fprintf(stderr, "usage: %s [OPTS]\n"
+		"\nOPTS:\n"
+		"    -d    detach program\n"
+		"    -f    lua script path\n"
+		"    -p    eBPF program path\n"
+		"    -i    iface\n"
+		"    -m    monitor\n"
+		"    -s    lua script\n",
+		prog);
+}
+
+static char *extract_lua_prog(const char *path)
+{
+	FILE *f;
+	long prog_size;
+	char *lua_prog;
+
+	f = fopen(path, "r");
+	if (f == NULL) {
+		perror("unable to xopen lua file");
+		return NULL;
+	}
+
+	fseek(f, 0 , SEEK_END);
+	prog_size = ftell(f);
+	rewind(f);
+
+	lua_prog = (char *) malloc(prog_size + 1);
+	memset(lua_prog, 0, prog_size + 1);
+	if (fread(lua_prog, 1, prog_size, f) < 0) {
```

Comment:
From https://github.com/luainkernel/linux/pull/3

```suggestion
	size_t read = fread(lua_prog, 1, prog_size, f);
	if (read < 0) {
```